operations/host_spawn.go

```diff
@@ -792,3 +796,367 @@ func hostRunCommand() cli.Command {
+func makeLocalParentDirs(local, remote string, pull bool) error {
+	var dirs string
+	if pull {
+		dirs = filepath.Dir(local)
+	} else {
+		dirs = filepath.Dir(remote)
+	}
+
+	return errors.Wrapf(os.MkdirAll(dirs, 0755), "could not create local parent directories '%s'", dirs)
+}
+
+// getUserAndHostname gets the user's spawn hosts and if the hostID matches one
+// of the user's spawn hosts, it returns the username and hostname for that
+// host.
+func getUserAndHostname(ctx context.Context, hostID, confPath string) (user, hostname string, err error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	conf, err := NewClientSettings(confPath)
+	if err != nil {
+		return "", "", errors.Wrap(err, "problem loading configuration")
+	}
+	client := conf.setupRestCommunicator(ctx)
+	defer client.Close()
+
+	params := model.APIHostParams{
+		UserSpawned: true,
+		Mine:        true,
+	}
+	hosts, err := client.GetHosts(ctx, params)
+	if err != nil {
+		return "", "", errors.Wrap(err, "problem getting your spawn hosts")
+	}
+
+	for _, h := range hosts {
+		if model.FromStringPtr(h.Id) == hostID {
+			catcher := grip.NewBasicCatcher()
+			user = model.FromStringPtr(h.User)
+			catcher.ErrorfWhen(user == "", "could not find login user for host '%s'", hostID)
+			hostname = model.FromStringPtr(h.HostURL)
+			catcher.ErrorfWhen(hostname == "", "could not find hostname for host '%s'", hostID)
+			return user, hostname, catcher.Resolve()
+		}
+	}
+	return "", "", errors.Errorf("could not find host '%s' in user's spawn hosts", hostID)
+}
+
+// sanityCheckRsync performs some basic sanity checks for the common case in
+// which you want to mirror two directories. The trailing slash means to
+// overwrite all of the contents of the destination directory, so we check that
+// they really want to do this.
+func sanityCheckRsync(localPath, remotePath string, pull bool) (ok bool, err error) {
+	localPathIsDir := strings.HasSuffix(localPath, string(os.PathSeparator))
+	remotePathIsDir := strings.HasSuffix(remotePath, string(os.PathSeparator))
+
+	if localPathIsDir && !pull {
+		ok, err := sanityPrompt(fmt.Sprintf("The local directory '%s' will overwrite any existing contents in the remote directory '%s'", localPath, remotePath))
+		if err != nil {
+			return false, errors.Wrap(err, "error while asking user to sanity check rsync command")
+		}
+		if !ok {
+			return false, nil
+		}
+	}
+	if remotePathIsDir && pull {
+		ok, err := sanityPrompt(fmt.Sprintf("The remote directory '%s' will overwrite any existing contents in the local directory '%s'", remotePath, localPath))
+		if err != nil {
+			return false, errors.Wrap(err, "error while asking user to sanity check rsync command")
+		}
+		if !ok {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// sanityPrompt prints the message and asks the user if they would like to
+// continue or exit.
+func sanityPrompt(msg string) (ok bool, err error) {
```

Comment:
is this different from `confirm`?
https://github.com/evergreen-ci/evergreen/blob/076d2b195781b1a791f0509548652a60049b228d/operations/prompt.go#L23

Reply:
No. I removed it.